Type errors in strict mode

Hello!

When using configify with the following two TS options:

```json
		"strictPropertyInitialization": false,
		"strict": true,
```

I get the following errors coming from the library itself:

```
node_modules/@itgorillaz/configify/src/configuration/resolvers/aws/parameter-store-configuration.resolver.ts:94:9 - error TS2322: Type 'unknown' is not assignable to type 'Error | undefined'.

94         error: e,
           ~~~~~

  node_modules/@itgorillaz/configify/src/configuration/resolvers/resolved-value.interface.ts:9:3
    9   error?: Error;
        ~~~~~
    The expected type comes from property 'error' which is declared here on type 'ResolvedValue'

node_modules/@itgorillaz/configify/src/configuration/resolvers/aws/secrets-manager-configuration.resolver.ts:96:9 - error TS2322: Type 'unknown' is not assignable to type 'Error | undefined'.

96         error: e,
           ~~~~~

  node_modules/@itgorillaz/configify/src/configuration/resolvers/resolved-value.interface.ts:9:3
    9   error?: Error;
        ~~~~~
    The expected type comes from property 'error' which is declared here on type 'ResolvedValue'

node_modules/@itgorillaz/configify/src/decorators/value.decorator.ts:43:3 - error TS2322: Type '(target: object, property: string) => void' is not assignable to type 'PropertyDecorator'.
  Types of parameters 'property' and 'propertyKey' are incompatible.
    Type 'string | symbol' is not assignable to type 'string'.
      Type 'symbol' is not assignable to type 'string'.

43   return (target: object, property: string) => {
     ~~~~~~

[10:57:33] Found 3 errors. Watching for file changes.
```

--- a/src/configuration/configuration.registry.ts
+++ b/src/configuration/configuration.registry.ts
@@ -28,7 +28,7 @@ export class ConfigurationRegistry {
    * @param {any}    target    the class target
    * @param {string} attribute the attribute name
    */
-  static registerAttribute(target: any, attribute: string): void {
+  static registerAttribute(target: any, attribute: string | symbol): void {
     (
       target[VALUE_PROPERTIES_METADATA] ||
       (target[VALUE_PROPERTIES_METADATA] = [])

--- a/src/configuration/resolvers/aws/parameter-store-configuration.resolver.ts
+++ b/src/configuration/resolvers/aws/parameter-store-configuration.resolver.ts
@@ -91,7 +91,7 @@ export class AwsParameterStoreConfigurationResolver
       return {
         id,
         key,
-        error: e,
+        error: e as Error,
         success: false,
       };
     }

--- a/src/configuration/resolvers/aws/secrets-manager-configuration.resolver.ts
+++ b/src/configuration/resolvers/aws/secrets-manager-configuration.resolver.ts
@@ -93,7 +93,7 @@ export class AwsSecretsManagerConfigurationResolver
       return {
         id,
         key,
-        error: e,
+        error: e as Error,
         success: false,
       };
     }

--- a/src/decorators/value.decorator.ts
+++ b/src/decorators/value.decorator.ts
@@ -40,7 +40,7 @@ export const Value = (
   key: string,
   options?: ValueOptions,
 ): PropertyDecorator => {
-  return (target: object, property: string) => {
+  return (target: object, property: string | symbol) => {
     ConfigurationRegistry.registerAttribute(target, property);
     Reflect.defineMetadata(VALUE_METADATA, { key, options }, target, property);
   };

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -12,6 +12,8 @@
     "baseUrl": "./",
     "incremental": true,
     "skipLibCheck": true,
+    "strict": true,
+    "strictPropertyInitialization": false,
     "strictNullChecks": true,
     "noImplicitAny": true,
     "strictBindCallApply": true,

